### PR TITLE
Fudge dice

### PR DESCRIPTION
**Pull request by Incoming5643**
They're six sided dice with two pluses, two minusii, and two blank sides.

They're also a jokey recipe you can create with any dice and a chocolate bar. Once you eat the literal fudge die you get a fudge die no matter what kind of die you used into its creation. This is INTENTIONAL and PART OF THE JOKE.

Adds them to the "rare dice" spawn in dice bags.

Formally defines a d6 as a separate thing from the generic parent die. the generic parent die is still a d6 for all intents and purposes except for the fact that it's now just called a generic "die" instead of a d6. This was so the recipe didn't list a d6 as needed for the fudge dice recipe when really any die would do.

Doesn't replace existent generic die with d6eses on the maps because I don't want this to conflict for a million years. I'll swap out the generic diii at some future point after this is added.

### CODE REVIEW

**Comment by Cheridan** [comment]
cute

**Comment by MrPerson** [comment]
What system are fudge dice use for again?

File: code/game/objects/items/weapons/dice.dm [hunk 1]
```diff
@@ -49,6 +51,16 @@
 	icon_state = "d4"
 	sides = 4
 
+/obj/item/weapon/dice/d6
+	name = "d6"
+
+/obj/item/weapon/dice/fudge
+	name = "fudge die"
+	desc = "A die with six sides but only three results. Is this a plus or a minus? Your mind is drawing a blank..."
+	sides = 3 //shhh
```
Comment (duncathan) [inline code/game/objects/items/weapons/dice.dm]: >3-sided die

Comment (Incoming5643) [inline code/game/objects/items/weapons/dice.dm]: it's a six sided die with each side represented twice. This ain't no barrel dice crap.

**Comment by MrPerson** [comment]
I question the practical use of these dice. Like literally, I can't imagine anyone using them in the next six months.

**Comment by Incoming5643** [comment]
I added a one sided dice, practicality is not an argument here.

**Comment by Niknakflak** [comment]
I see someone play DnD with dice about once a year or so.

**Comment by Incoming5643** [comment]
Full disclosure I want to do fun stuff with dice down the line*, so I have a vested interest in adding <i>wacky dice</i>™ now.

Also come on, it's funny.

* "down the line" for me is potentially a very long timescale

**Comment by MrPerson** [comment]
I can see someone looking at the set of dice and planning something fun with them. I can even see that with the coin. I don't see that ever happening with the fudge dice. Besides, you can replicate this with a d6 anyway. 1 or 2 is -, 3 or 4 is blank, 5 or 6 is +. Do we really need special dice just for this?

Now the code itself is cool. I'm interested in what other special dice you wanted to do.

**Comment by Incoming5643** [comment]
We've had this discussion before, when I was adding the last set. The only dice that can be "missing" from the bag is the slot the d00 used to fill, which the d10 can replicate. The slot is for the "auxiliary/redundant/jokey dice". 

If I add enough dice I'll probably sprite some sort of larger dice container that just comes with a buttload of all the dice you can order from cargo.

**Comment by Supermichael777** [comment]
still want a chaos die for wiznerds. 
God may not play dice with the universe but you sure as hell do!

**Comment by Incoming5643** [comment]
One of the ideas is a magical dice bag where each die has a different set of effects depending on the roll. You'd take a die out of the bag, throw it/roll it yourself, and then it'd cease to be magic after having its effect. The bag would regenerate dice over time, but you'd have no control over which dice you got. I've had a half finished pull of this for months now, I just ran out of ideas for effects before I ran out of dice. The fudge dice was going to be size modifiers.

Anyway that's not this, just something waiting in the wings maybe.

**Comment by Bawhoppen** [comment]
:game_die:

**Comment by Cheridan** [comment]
its cute

**Comment by Chocobro** [comment]
did you seriously make a fucking one-sided dice
that's some bluespace shit

**Comment by Bawhoppen** [comment]
one sided dice exist irl you realise

**Comment by duncathan** [comment]
For instance, the yellow solid pool ball is effectively a one-sided die

**Comment by Incoming5643** [comment]
shit now I want to make the one sided die literally that pool ball

**Comment by Chocobro** [comment]
@Bawhoppen it's called a joke???????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????

**Comment by Bawhoppen** [comment]
@Chocobro :question:

File: code/game/objects/items/weapons/dice.dm [hunk 2]
```diff
@@ -18,14 +19,15 @@
 	new /obj/item/weapon/dice/d20(src)
 	if(special_die == "100")	new /obj/item/weapon/dice/d100(src)
 
-/obj/item/weapon/dice
-	name = "d6"
+/obj/item/weapon/dice //depreciated d6, use /obj/item/weapon/dice/d6 if you actually want a d6
```
Comment (JJRcop) [inline code/game/objects/items/weapons/dice.dm]: deprecate

Comment (duncathan) [inline code/game/objects/items/weapons/dice.dm]: Poor d6, nobody appreciates him :c

**Pull request merged**